### PR TITLE
Run tests on more operating systems

### PR DESCRIPTION
**Pull request by Etiene**
* Adds docker containers for Amazon Linux and CentOS
* Updates Ubuntu container from 14.04 to 16.04

### CODE REVIEW

File: test/amazonlinux/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM amazonlinux
```
Comment (Etiene) [inline test/amazonlinux/Dockerfile]: Should I lock this at 2017.12 or leave it at latest?
(I'm inclined to locking)

Comment (brikis98) [reply test/amazonlinux/Dockerfile]: I think latest is fine, as we want to know that new changes in Amazon Linux don't break our code. In the future, we'll prob add support for Amazon Linux 2.

Comment (Etiene) [reply test/amazonlinux/Dockerfile]: * after discussion on slack, leaving at `latest^

**Comment by Etiene** [comment]
Oh yes, of course :)

**Pull request merged**